### DISCUSSION TITLE
Bazel

### DISCUSSION THREAD

**Discussion by DTSHNICK** [COVESA/vsomeip#649] [Q&A] [answered]
Hello
I plan to add this library to my project, but my build system is working on Bazel.
Is there any hint how I can rewrite CMake instructions on Bazel rules?

My main problem is that I can't understand how I should include the headers files that are included in the code like
`#include <vsomeip/primitive_types.hpp>`

**Comment by phongt** [comment, marked as answer]
Hi you can reference Bazel solution from this https://github.com/ApexAI/vsomeip/tree/release-3.4.10-apex

**Reply by DTSHNICK** [reply]
Thank you @phongt 
I will check it

**Reply by DTSHNICK** [reply]
It works!

**Reply by goncaloalmeida** [reply]
@phongt why not create a PR with this bazel build in the main repository?

**Reply by phongt** [reply]
>why not create a PR with this bazel build in the main repository?

@goncaloalmeida I can not answer you. I only found it in my searching from internet and not maintainer the repo .

**Reply by goncaloalmeida** [reply]
thanks. we will check that then
